package/MDAnalysis/topology/TOPParser.py

```diff
@@ -563,9 +586,16 @@ def parsesection_mapper(self, numlines, mapper):
             A list of all entries in a given parm7 section
         """
         section = []
-        y = next(self.topfile).strip("%FORMAT(")
-        y.strip(")")
-        x = FORTRANReader(y)
+
+        def get_fmt(file):
+            if (line := next(file))[:7] == "%FORMAT":
```

Comment:
What does the syntax `(line := ...)` do, i.e., `(x := y)` ??

Reply:
I guess it is equivalent to 
```python
line = next(file)
if line[:7] == "%FORMAT":
   ...
```
I hadn't see C-style "assignment inside conditionals" in Python. In which version of Python was this introduced?

Reply:
Personally, I find the old-school Python two-line syntax clearer but I'll concede that this is a matter of personal style. More important is if this syntax is supported by all our Python versions.

Reply:
Yes, the 'walrus operator', added on 3.8 and the minimum supported one is 3.9, so we'd be good. Anyways, I ended up dropping it.